### PR TITLE
hubble: move hubble-serve out of daemon, re-organize packages

### PR DESCRIPTION
**Pull request by rolinh**
This commit moves package hubble-serve out of daemon/ and into
pkg/hubble/server. The package previous pkg/hubble/server is renamed to
pkg/hubble/observer to avoid possible confusion with hubble's gRPC
server. The observer package is indeed an implementation of the observer
server gRPC service. In order to avoid confusion of the observer
implementation and its protobuf definition, modules that imported the
protobuf definition have been import-renamed to observerpb.

This commit is in preparation of the addition of the peer service
implementation and brings no functional changes.

### CODE REVIEW

File: pkg/hubble/observer/local_observer.go [hunk 1]
```diff
@@ -78,21 +77,21 @@ type LocalObserverServer struct {
 	log *logrus.Entry
 
 	// channel to receive events from observer server.
-	eventschan chan *observer.GetFlowsResponse
+	eventschan chan *observerpb.GetFlowsResponse
```
Comment (gandro) [inline pkg/hubble/observer/local_observer.go:81]: nit: this field can actually be removed. it is not used anymore in embedded Hubble.

File: pkg/hubble/observer/option.go [hunk 2]
```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package serveroption
```
Comment (gandro) [inline pkg/hubble/observer/option.go]: Pinging @glibsm since I believe he had a rationale when he introduced this as a separate package.

**Review by gandro — COMMENTED** [review]
Love it! I want to give others a chance to voice their opinion as well though.

File: pkg/hubble/observer/local_observer.go (hunk 1 above)
Comment (rolinh) [reply pkg/hubble/observer/local_observer.go:81]: Sure but I don't think it should be part of this PR. This is just code re-org/renaming without any other changes.

**Review by michi-covalent — COMMENTED** [review]
lgtm let's wait for @glibsm

File: pkg/hubble/observer/option.go (hunk 2 above)
Comment (glibsm) [reply pkg/hubble/observer/option.go]: `serveroption` was moved into a separate package so other packages providing options don't pull in the entire dependency tree of the server

Comment (rolinh) [reply pkg/hubble/observer/option.go]: I moved all options (observer, server, proxy) to option packages.

**Review by michi-covalent — APPROVED** [review]

**Comment by Rolinh** [comment]
test-me-please

**Comment by Rolinh** [comment]
test-with-kernel

**Comment by Rolinh** [comment]
test-with-kernel

**Comment by Rolinh** [comment]
test-with-kernel

**Review by tklauser — APPROVED** [review]
LGTM

**Pull request merged**